apps/condo/domains/meter/hooks/useImporterFunctions.ts

```diff
@@ -207,83 +221,65 @@ export const useImporterFunctions = (): [Columns, RowNormalizer, RowValidator, O
-                organization: { id: userOrganizationIdRef.current },
-                address_i: suggestion.value,
-            }, undefined)
-
-            const propertyId = !isEmpty(properties) ? get(properties[0], 'id') : null
-            const propertyMap = !isEmpty(properties) ? get(properties[0], 'map') : null
-            if (!propertyId) {
-                return { row, addons }
-            }
+            addons.readingSubmissionDate = parseDateOrMonth(readingSubmissionDate)
+        } catch (e) {
+            addons.invalidReadingSubmissionDate = true
+        }
 
-            addons.propertyId = propertyId
-            addons.propertyMap = propertyMap
-            addons.unitType = UNIT_TYPE_TRANSLATION_TO_TYPE[String(unitType).toLowerCase()]
-
-            const searchMeterWhereConditions = {
-                organization: { id: userOrganizationIdRef.current },
-                property: { id: propertyId },
-                unitName,
-                unitType: addons.unitType,
-                accountNumber,
-                number: meterNumber,
-            }
+        // Current suggestion API provider returns no suggestions for address with flat number
+        const suggestionOptions = await addressApi.getSuggestions(String(address))
+        const suggestion = get(suggestionOptions, ['suggestions', 0])
 
-            const meterOptions = await searchMeter(client, searchMeterWhereConditions, SortMetersBy.CreatedAtDesc)
-            addons.meterId = meterOptions.length > 0 ? meterOptions[0].value : null
+        if (!suggestion) return { row, addons }
 
-            const METER_RESOURCE_ABBREVIATION_TO_ID = {
-                [HotWaterResourceTypeValue]: HOT_WATER_METER_RESOURCE_ID,
-                [ColdWaterResourceTypeValue]: COLD_WATER_METER_RESOURCE_ID,
-                [ElectricityResourceTypeValue]: ELECTRICITY_METER_RESOURCE_ID,
-                [HeatSupplyResourceTypeValue]: HEAT_SUPPLY_METER_RESOURCE_ID,
-                [GasSupplyResourceTypeValue]: GAS_SUPPLY_METER_RESOURCE_ID,
-            }
-            addons.meterResourceId = METER_RESOURCE_ABBREVIATION_TO_ID[String(meterResourceTypeAbbr)]
+        addons.address = suggestion.value
 
-            try {
-                addons.readingSubmissionDate = parseDateOrMonth(readingSubmissionDate)
-            } catch (e) {
-                addons.invalidReadingSubmissionDate = true
-            }
-        } catch (error) {
-            addons.invalidNormalization = true
-            console.error('meterReadingNormalizer error')
-            console.error(error)
-            console.error(row)
+        // Used tell whether suggestion API has found specified address at all
+        const properties = await searchPropertyWithMap(client, {
+            organization: { id: userOrganizationIdRef.current },
+            address_i: suggestion.value,
+        }, undefined)
+
+        const propertyId = !isEmpty(properties) ? get(properties[0], 'id') : null
```

Comment:
it seems like this will work too
```suggestion
        const propertyId = get(properties[0], 'id',  null)
```

Reply:
I just moved the existing code)

Reply:
I’ve already approved it, correct it if you want, leave it if you don’t) I don’t block pull requests